python/src/examples/example4/chattool.py

```diff
@@ -0,0 +1,82 @@
+        self.messageEntry.setObjectName("messageEntry")
+        self.sendButton = QtWidgets.QPushButton(self.centralwidget)
+        self.sendButton.setGeometry(QtCore.QRect(470, 660, 121, 41))
+        self.sendButton.setObjectName("sendButton")
+        self.verticalLayoutWidget = QtWidgets.QWidget(self.centralwidget)
+        self.verticalLayoutWidget.setGeometry(QtCore.QRect(10, 20, 581, 631))
+        self.verticalLayoutWidget.setObjectName("verticalLayoutWidget")
+        self.verticalLayout = QtWidgets.QVBoxLayout(self.verticalLayoutWidget)
+        self.verticalLayout.setContentsMargins(0, 0, 0, 0)
+        self.verticalLayout.setObjectName("verticalLayout")
+        self.chatHistory = QtWidgets.QGroupBox(self.centralwidget)
+        self.chatHistory.setGeometry(QtCore.QRect(600, 10, 361, 701))
+        self.chatHistory.setObjectName("chatHistory")
+        MainWindow.setCentralWidget(self.centralwidget)
+        self.menubar = QtWidgets.QMenuBar(MainWindow)
+        self.menubar.setGeometry(QtCore.QRect(0, 0, 965, 21))
+        self.menubar.setObjectName("menubar")
+        self.menuServer = QtWidgets.QMenu(self.menubar)
+        self.menuServer.setObjectName("menuServer")
+        self.menuSettings = QtWidgets.QMenu(self.menubar)
+        self.menuSettings.setObjectName("menuSettings")
+        self.menuRSA = QtWidgets.QMenu(self.menubar)
+        self.menuRSA.setObjectName("menuRSA")
+        MainWindow.setMenuBar(self.menubar)
+        self.statusbar = QtWidgets.QStatusBar(MainWindow)
+        self.statusbar.setObjectName("statusbar")
+        MainWindow.setStatusBar(self.statusbar)
+        self.actionChange_settings = QtWidgets.QAction(MainWindow)
+        self.actionChange_settings.setObjectName("actionChange_settings")
+        self.actionInfo = QtWidgets.QAction(MainWindow)
+        self.actionInfo.setObjectName("actionInfo")
+        self.menuServer.addAction(self.actionInfo)
+        self.menuSettings.addAction(self.actionChange_settings)
+        self.menubar.addAction(self.menuSettings.menuAction())
+        self.menubar.addAction(self.menuServer.menuAction())
+        self.menubar.addAction(self.menuRSA.menuAction())
+
+        self.retranslateUi(MainWindow)
+        QtCore.QMetaObject.connectSlotsByName(MainWindow)
+
+    def retranslateUi(self, MainWindow):
+        _translate = QtCore.QCoreApplication.translate
+        MainWindow.setWindowTitle(_translate("MainWindow", "RSA - Chattool"))
+        self.sendButton.setText(_translate("MainWindow", "Send Message"))
+        self.chatHistory.setTitle(_translate("MainWindow", "Keys"))
+        self.menuServer.setTitle(_translate("MainWindow", "Server"))
+        self.menuSettings.setTitle(_translate("MainWindow", "Settings"))
+        self.menuRSA.setTitle(_translate("MainWindow", "RSA"))
+        self.actionChange_settings.setText(_translate("MainWindow", "Change settings"))
+        self.actionInfo.setText(_translate("MainWindow", "Info"))

```

Comment:
```suggestion
    def changeEvent(self, event):
        super().changeEvent(event)
        if isinstance(event, QtCore.QEvent):
            ui.retranslateUi(self)
```
Dies sollte funktionieren, getestet habe ich es aber nicht.

Wird eine neue Übersetzung installiert, so wird das ensprechende Event erkannt und die Funktion `retranslateUi` der GUI aufgerufen.